src/actions/receive-profile.js

```diff
@@ -802,3 +823,160 @@ export function retrieveProfileFromFile(
     }
   };
 }
+
+/**
+ * This action retrieves several profiles and push them into 1 profile using the
+ * information contained in the query.
+ */
+export function retrieveProfilesToCompare(
+  profileViewUrls: string[]
+): ThunkAction<Promise<void>> {
+  return async dispatch => {
+    dispatch(waitingForProfileFromUrl());
+
+    try {
+      // First we get a state from each URL. From these states we'll get all the
+      // data we need to fetch and process the profiles.
+      const profileStates = profileViewUrls.map(url =>
+        stateFromLocation(new URL(url))
+      );
+      const hasSupportedDatasources = profileStates.every(
+        state => state.dataSource === 'public'
+      );
+      if (!hasSupportedDatasources) {
+        throw new Error(
+          'Only public uploaded profiles are supported by the comparison function.'
+        );
+      }
+
+      // Then we retrieve the profiles from the online store, and unserialize
+      // and process them if needed.
+      const promises = profileStates.map(async ({ hash }) => {
+        const profileUrl = getProfileUrlForHash(hash);
+        const response = await _fetchProfile({
+          url: profileUrl,
+          onTemporaryError: (e: TemporaryError) => {
+            dispatch(temporaryError(e));
+          },
+        });
+        const serializedProfile = response.profile;
+        if (!serializedProfile) {
+          throw new Error('Expected to receive a profile from _fetchProfile');
+        }
+
+        const profile = unserializeProfileOfArbitraryFormat(serializedProfile);
+        return profile;
+      });
+
+      // Once all profiles have been fetched and unserialized, we can start
+      // pushing them to a brand new profile. This resulting profile will keep
+      // only the 2 selected threads from the 2 profiles.
+      const profiles = await Promise.all(promises);
+      const resultProfile = getEmptyProfile();
```

Comment:
I split the function but didn't move the existing tests I wrote in this PR. I think I'll indeed write more tests as I move forward implementing the merge, and this will make it easier to have more targeted tests in the future.